watcherx/file.go

```diff
@@ -0,0 +1,118 @@
+package watcherx
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/pkg/errors"
+)
+
+func WatchFile(ctx context.Context, file string, c EventChannel) error {
+	w, err := fsnotify.NewWatcher()
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	dir := path.Dir(file)
+	if err := w.Add(dir); err != nil {
+		return errors.WithStack(err)
+	}
+	realFile, err := filepath.EvalSymlinks(file)
+	if err != nil {
+		if _, ok := err.(*os.PathError); ok {
+			// the file watcher should still watch the directory to get notified for
+			// file creation
+			realFile = ""
+		} else {
+			return errors.WithStack(err)
+		}
```

Comment:
Best-practice Go code follows a pattern of "early return": https://dave.cheney.net/practical-go/presentations/qcon-china.html#_return_early_rather_than_nesting_deeply

```suggestion
			realFile = ""
		}
		return errors.WithStack(err)
```